src/ipc/mod.rs

```diff
@@ -149,9 +263,9 @@ impl<'a, T: Ipc> Backend<'a, T> {
     }
 }
 
-impl<'a, T: Ipc> Drop for Backend<'a, T> {
+impl<T: Ipc> Drop for SingleBackend<T> {
     fn drop(&mut self) {
-        Rc::get_mut(&mut self.sock)
+        Arc::get_mut(&mut self.sock)
```

Comment:
why does this have to be an `Arc`?

Reply:
because backend is moved inside the sending thread (around line 128) so it has to be send which means all of its members must be send